block/s3/inventory_test.go

```diff
@@ -149,17 +161,56 @@ func TestIterator(t *testing.T) {
 	}
 }
 
-type mockParquetReader struct {
-	rows    []*s3.ParquetInventoryObject
+type mockInventoryReader struct {
+	openFiles map[string]bool
+}
+
+type mockInventoryFileReader struct {
+	rows    []*inventorys3.InventoryObject
 	nextIdx int
+	mgr     *mockInventoryReader
+	key     string
+}
+
+func (m *mockInventoryFileReader) MinValue() string {
+	if len(m.rows) == 0 {
+		return ""
```

Comment:
the MinValue/MaxValue of empty files is not really an issue since it doesn't matter where they come in ordering of the iterator - they are empty so no values will be returned.

Reply:
But these _are_ used in sorting, and will potentially fail the non-overlapping test.

Agree this is merely confusing, since we do control the test.